tools/dispatch_data.py

```diff
@@ -0,0 +1,116 @@
+"""Launching distributed graph partitioning pipeline """
+import os
+import sys
+import argparse
+import logging
+import json
+
+INSTALL_DIR = "/home/ubuntu/workspace/DistDgl-Utils-updatedfileformat"
+LAUNCH_SCRIPT = "distgraphlaunch.py"
+PIPELINE_SCRIPT = "data_proc_pipeline.py"
+
+UDF_WORLD_SIZE = "world-size"
+UDF_PART_DIR = "partitions-dir"
+UDF_INPUT_DIR = "input-dir"
+UDF_GRAPH_NAME = "graph-name"
+UDF_SCHEMA = "schema"
+UDF_NUM_PARTS = "num-parts"
+UDF_OUT_DIR = "output"
+
+LARG_PROCS_MACHINE = "num_proc_per_machine"
+LARG_IPCONF = "ip_config"
+LARG_MASTER_PORT = "master_port"
+
+def get_launch_cmd(args, user_python_exe) -> str: 
+
+    cmd = pythonpath=os.environ.get("PYTHONPATH", "") + get_python_path(user_python_exe) + " " + os.path.join(INSTALL_DIR, LAUNCH_SCRIPT)
+    cmd = f"{cmd} --{LARG_PROCS_MACHINE} 1 "
```

Comment:
This could be exposed as the main argument of `dispatch_data.py` with a proper default value. One question. Say we have 4 partitions. Is it true that the world size must be 4 but `num_proc_per_machine` could be arbitrary? In other words, any restriction on this?

Reply:
I think it must be 1.